api/src/main/java/jakarta/servlet/http/Cookie.java

```diff
@@ -422,4 +423,103 @@ public void setHttpOnly(boolean isHttpOnly) {
+        this.sameSiteMode = sameSiteMode;
+    }
+
+    /**
+     * Returns the <i>SameSite</i>> mode associated with this cookie.
+     *
+     * @return The <i>SameSite</i> mode associated with this cookie. May be null.
+     *
+     * @since Servlet 4.x
+     */
+    public SameSiteMode getSameSiteMode() {
+        return sameSiteMode;
+    }
+
+    /**
+     * The available values that can be used with the cookie's <i>SameSite</i> attribute.
+     */
+    public enum SameSiteMode {
+
+        /**
+         * This mode effectively disables the protection provided by SameSite cookies.
+         *
+         * <p>
+         *     When SameSiteMode is {@literal None}, applications should have additional measures to protect cookies
+         *     from CSRF attacks.
+         * </p>
+         *
+         * <p>
+         *     Browsers may default to {@link SameSiteMode#LAX} or {@link SameSiteMode#STRICT} when the flag is not
+         *     explicitly set by applications in the Set-Cookie header. Setting the {@literal SameSite} flag to
+         *     {@literal None} ensures that this default behavior is disabled in browsers.
+         * </p>
+         *
+         * <p>
+         *     Note that this mode is browser dependent, and browsers may require cookies to have the {@literal Secure}
+         *     flag to be set at the same time.
+         * </p>
+         */
+        NONE("None"),
+        /**
+         * This mode allows sending cookies along for cross-site top level navigation requests.
+         *
+         * <p>
+         *     When SameSiteMode is {@literal Lax}, the browser is allowed to send the cookie value along with top level
+         *     navigation requests when making requests cross-site.
+         * </p>
+         */
+        LAX("Lax"),
+        /**
+         * This mode prevents browsers from sending cookies along with any kind of cross-site requests.
+         *
+         * <p>
+         *     When SameSiteMode is {@literal Strict}, the browser is not allowed to send the cookie value along with
+         *     cross-site requests.
+         * </p>
+         *
+         * <p>
+         *     This behaviour could prevent applications seeing the end-user's session when a cross-site navigation
+         *     takes place. To overcome this, applications should consider utilising multiple cookies to keep track of
+         *     sessions.
+         * </p>
+         */
+        STRICT("Strict");
+
+        private final String value;
+
+        SameSiteMode(String value) {
+            this.value = value;
+        }
+
+        /**
+         * Retrieves the {@link SameSiteMode} instance from its String representation.
+         *
+         * @param value The String to be converted.
+         * @return The corresponding {@link SameSiteMode} enum.
+         */
+        public static SameSiteMode fromString(String value) {
+            return valueOf(value.toUpperCase());
```

Comment:
This needs to avoid using default locale.

Proposed change:
``` java
return valueOf(value.toUpperCase(Locale.ENGLISH));
```

Reply:
Fixed